src/riak_ql.yrl.tests

```diff
@@ -3,8 +3,13 @@
 
 -define(sql_compilation_assert(String, Expected),
 	Toks = riak_ql_lexer:get_tokens(String),
-	{ok, Got} = parse(Toks),
-	?assertEqual(Expected, Got)).
+	Got = parse(Toks),
+	?assertEqual({ok, Expected}, Got)).
+
+-define(sql_compilation_fail(QL_string),
+	Toks = riak_ql_lexer:get_tokens(QL_string),
+	Got = parse(Toks),
+	?assertMatch({error, _}, Got)).
```

Comment:
We haven't yet got the whole system returning errors to the end user - we should probably reconsider the error messages that the lexer/parsers emit in the context of the errors that the query validator and query compiler emit back to the user.

I will add an issue on my issue list to come back to this.